lib/fluent/plugin/in_tail.rb

```diff
@@ -718,6 +720,7 @@ def register_watcher(watcher)
       end
 
       def detach
+        @io_handler.ready_to_shutdown if @io_handler
```

Comment:
`detach` is called not only when shutdown but also on refreshing watchers.
But I think we don't need consider the later case because it will occur only when idling case (not occur while running `on_notify`).